### PR TITLE
nip-37: addressable and replaceable events change.

### PR DESCRIPTION
**Pull request by kehiy**

### CODE REVIEW

File: 37.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Relays MAY reject requests for non-replaceable or non-addressable events or any not acceptable filter and respond with a NOTICE message.
+
+## Examples
+
+### Single profile update check
+
+```jsonc
+["COUNT", <subscription_id>, {"kinds": [0], "pubkey": "badbdda507572b397852048ea74f2ef3ad92b1aac07c3d4e1dec174e8cdc962a"}]
+["COUNT", <subscription_id>, 1724958773]
```
Comment (AsaiToshiya) [inline 37.md:53]:
```suggestion
["CHANGED", <subscription_id>, {"kinds": [0], "pubkey": "badbdda507572b397852048ea74f2ef3ad92b1aac07c3d4e1dec174e8cdc962a"}]
["CHANGED", <subscription_id>, 1724958773]
```

File: 37.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+### Multiple profile update check
+
+```jsonc
+["COUNT", <subscription_id>, {"kinds": [0], "pubkey": "badbdda507572b397852048ea74f2ef3ad92b1aac07c3d4e1dec174e8cdc962a"}, {"kinds": [0], "pubkey": "bd4ae3e67e29964d494172261dc45395c89f6bd2e774642e366127171dfb81f5"}, {"kinds": [0], "pubkey": ""}]
+["COUNT", <subscription_id>, 1724958773, 1727247549, 0]
```
Comment (AsaiToshiya) [inline 37.md:60]:
```suggestion
["CHANGED", <subscription_id>, {"kinds": [0], "pubkey": "badbdda507572b397852048ea74f2ef3ad92b1aac07c3d4e1dec174e8cdc962a"}, {"kinds": [0], "pubkey": "bd4ae3e67e29964d494172261dc45395c89f6bd2e774642e366127171dfb81f5"}, {"kinds": [0], "pubkey": ""}]
["CHANGED", <subscription_id>, 1724958773, 1727247549, 0]
```

**Review by AsaiToshiya — COMMENTED** [review]
What will be the results of the filter that return multiple events?

**Comment by fiatjaf** [comment]
This is an interesting proposal that could fix Nostr.

**Comment by kehiy** [comment]
@AsaiToshiya based on what i defined here relay can refuse to respond or just put a 0 as a result there.

**Comment by kehiy** [comment]
@fiatjaf, when they are online and connected, they will receive new updates. but when you open the client, you have a bunch of cached profiles, relat list metadata, and...

you (client) want to update them, but your state may be up to date. So you are forced to request whole events from relay(s) and check if they are newer than cached ones. using `CHANGED`, you can check if you are up to date without wasting bandwidth. in one message, you will check how many of your cached events are up to date. you will keep them and request new ones from relays.

**Comment by staab** [comment]
I don't think this solves anything that negentropy doesn't also solve, and in a more general purpose way.

**Comment by mikedilger** [comment]
You can currently say:   [filter1: id1, pubkey1, dtag1, >timestamp1], [filter2: id2, pubkey2, dtag2, >timestamp2]  and get the same thing.  REQ allows multiple filters, and you just want to say "give me an event if it is newer than the one I have".  If it wasn't newer, you didn't waste any bandwidth.

**Comment by kehiy** [comment]
@mikedilger makes sense. i'm going to close it.

**Pull request closed without merging**